resources/assets/js/components/JobBuilderSkills/JobBuilderSkillsPage.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { useEffect, useState } from "react";
+import { InjectedIntlProps, FormattedMessage, injectIntl } from "react-intl";
+import { connect } from "react-redux";
+import ReactDOM from "react-dom";
+import RootContainer from "../RootContainer";
+import { Job, JobPosterKeyTask, Criteria, Skill } from "../../models/types";
+import {
+  jobBuilderIntroProgressState,
+  jobBuilderDetailsProgressState,
+  jobImpactProgressState,
+  jobBuilderEnvProgressState,
+  VALID_COUNT,
+  jobTasksProgressState,
+} from "../JobBuilder/jobBuilderHelpers";
+import {
+  progressTrackerLabels,
+  progressTrackerTitles,
+} from "../JobBuilder/jobBuilderMessages";
+import ProgressTracker from "../ProgressTracker/ProgressTracker";
+import JobBuilderSkills from "./JobBuilderSkills";
+import { managerJobIndex } from "../../helpers/routes";
+import { ProgressTrackerItem } from "../ProgressTracker/types";
+import { RootState } from "../../store/store";
+import {
+  getJob,
+  getTasksByJob,
+  getCriteriaByJob,
+} from "../../store/Job/jobSelector";
+import { getSkills } from "../../store/Skill/skillSelector";
+import { DispatchType } from "../../configureStore";
+import { fetchJob, fetchJobTasks, fetchCriteria, batchUpdateCriteria } from "../../store/Job/jobActions";
+import { fetchSkills } from "../../store/Skill/skillActions";
+
+interface JobBuilderSkillsPageProps {
+  jobId: number;
+  job: Job | null;
+  loadJob: (jobId: number) => Promise<void>;
+  skills: Skill[];
+  loadSkills: () => Promise<void>;
+  keyTasks: JobPosterKeyTask[];
+  loadTasks: (jobId: number) => Promise<void>;
+  criteria: Criteria[];
+  loadCriteria: (jobId: number) => Promise<void>;
+  handleSubmitCriteria: (
+    jobId: number,
+    criteria: Criteria[],
+  ) => Promise<Criteria[]>;
+}
+
+const JobBuilderSkillsPage: React.FunctionComponent<
+  JobBuilderSkillsPageProps & InjectedIntlProps
+> = ({
+  jobId,
+  job,
+  loadJob,
+  skills,
+  loadSkills,
+  keyTasks,
+  loadTasks,
+  criteria,
+  loadCriteria,
+  handleSubmitCriteria,
+  intl,
+}): React.ReactElement => {
+  // Trigger fetching of jobs, skills, tasks, and criteria on first load, or when jobId changes
+  const [isLoadingJob, setIsLoadingJob] = useState(false);
+  useEffect((): void => {
```

Comment:
We might want to look into writing our own reusable hooks for this kind of thing moving forward:

https://reactjs.org/docs/hooks-custom.html#extracting-a-custom-hook

Reply:
Ah, you're right, this looks like it could be a model use case.